src/plugins/usage_collection/server/collector/collector_set.test.ts

```diff
@@ -8,18 +8,25 @@
 
 import { noop } from 'lodash';
 import { Collector } from './collector';
-import { CollectorSet } from './collector_set';
+import { CollectorSet, CollectorSetConfig } from './collector_set';
 import { UsageCollector } from './usage_collector';
 
 import {
   elasticsearchServiceMock,
   loggingSystemMock,
   savedObjectsClientMock,
   httpServerMock,
+  executionContextServiceMock,
 } from '../../../../core/server/mocks';
 
 describe('CollectorSet', () => {
   const logger = loggingSystemMock.createLogger();
+  const executionContext = executionContextServiceMock.createSetupContract();
+
+  const collectorSetConfig: CollectorSetConfig = {
+    logger,
+    executionContext,
+  };
```

Comment:
NIT: doesn't seem strictly necessary in that suite, but it's usually a good practice to cleanup the test's sandbox between each test by recreating all the dependencies.

```ts
let logger: ReturnType<typeof loggingSystemMock.createLogger>;
let executionContext: ReturnType<typeof executionContextServiceMock.createSetupContract>;
let collectorSetConfig: CollectorSetConfig;

beforeEach(() => {
  logger = loggingSystemMock.createLogger();
  executionContext = executionContextServiceMock.createSetupContract();
  collectorSetConfig: CollectorSetConfig = {
    logger,
    executionContext,
  };
})
```

This would also allow to remove the `loggerSpies` work-around to reset the calls to `log.XXX`